### ISSUE TITLE
when 2.2 node join 2.1 node it will coredump

### ISSUE THREAD

**Issue by wqwiii** [scylladb/scylladb#3871]
```
Oct 19 17:51:09 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 6] seastar_memory - oversized allocation: 3145728 bytes, please report: at   0x73033d#012  0x736ad6#012  0x737034#012  0x818d62#012  0x2d30090#012  0x592ee6#012  0x5930a1#012  0x658ca7#012  0x65f593#012  0x72eaed#012  /lib64/libpthread.so.0+0x7dc4#012  /lib64/libc.so.6+0xf776c
Oct 19 17:51:11 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 6] rpc - client 10.80.176.39: request size 2570647147 large than memory limit 312056217
Oct 19 17:51:11 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 0] stream_session - [Stream #65001c50-d384-11e8-8290-000000000002] Received failed complete message, peer=10.80.176.39
Oct 19 17:51:11 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 0] stream_session - [Stream #65001c50-d384-11e8-8290-000000000002] Streaming plan for Bootstrap-click-index-1 failed, peers={10.80.176.39}, tx=0 KiB, 0.00 KiB/s, rx=557786 KiB, 11203.00 KiB/s
Oct 19 17:51:11 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 0] range_streamer - Bootstrap with 10.80.176.39 for keyspace=click failed, took 241 seconds: streaming::stream_exception (Stream failed)
Oct 19 17:51:11 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 0] range_streamer - Bootstrap failed, took 241 seconds, nr_ranges_remaining=370
Oct 19 17:51:11 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 0] range_streamer - Bootstrap failed to stream. Will retry in 90 seconds ...
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: Segmentation fault on shard 2.
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: Backtrace:
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b010c
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b03b5
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b0403
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libpthread.so.0+0x000000000000f36f
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000183b9ae
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000018b9b32
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x0000000000e0f2bd
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000010e6293
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000010e8d1e
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b3c6b
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005916cb
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000058e984
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x0000000000658c6e
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000065f593
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000072eaed
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libpthread.so.0+0x0000000000007dc4
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libc.so.6+0x00000000000f776c
Oct 19 17:51:12 iZbp14rlw7a7q9lu2ccp0lZ kernel: reactor-2[5763]: segfault at 1 ip 000000000183b9af sp 00007f5da51d4700 error 4 in scylla[400000+46fc000]
```

**Comment by wqwiii** [comment]
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 0] range_streamer - Bootstrap failed, took 124 seconds, nr_ranges_remaining=513
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: [shard 0] range_streamer - Bootstrap failed to stream. Will retry in 60 seconds ...
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: scylla: ./seastar/core/future.hh:345: void seastar::future_state<>::set_exception(std::__exception_ptr::exception_ptr): Assertion `_u.st == state::future' failed.
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: Aborting on shard 1.
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: Backtrace:
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b010c
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b03b5
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b0463
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libpthread.so.0+0x000000000000f36f
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libc.so.6+0x00000000000351d6
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libc.so.6+0x00000000000368c7
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libc.so.6+0x000000000002e145
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libc.so.6+0x000000000002e1f1
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x0000000000eb50b8
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000183b9b0
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000018b9b32
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x0000000000e0f2bd
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000010e6293
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000010e8d1e
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005b3c6b
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x00000000005916cb
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000058e984
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x0000000000658c6e
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000065f593
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: 0x000000000072eaed
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libpthread.so.0+0x0000000000007dc4
Oct 19 17:55:50 iZbp14rlw7a7q9lu2ccp0lZ scylla: /lib64/libc.so.6+0x00000000000f776c

**Comment by tzach** [comment]
@wqwiii can you please upload the core dump as described here:
https://docs.scylladb.com/troubleshooting/report_scylla_problem/#core-dump

**Comment by slivne** [comment]
@wqwiii - are you trying to increase/enlarge the cluster size by adding a new node with a higher version ?

we do not support topology_changes in the middle of an upgrade - please either:
- add a node of version 2.1 - while your cluster is 2.1
- upgrade the existing cluster to 2.2 and then add a new node of 2.2

**Comment by slivne** [comment]
@asias 
- Can we validate that a new node is added with the same version the cluster is running
- Can we also validate that a cluster is under going repair/streaming - and not allow an upgrade ?

thoughts ?

**Comment by asias** [comment]
@wqwiii Can you please at least provide the exact scylla version number you were using 2.1.X and 2.2.Y.

**Comment by asias** [comment]
> @asias
> 
>     * Can we validate that a new node is added with the same version the cluster is running

We do not require this when adding a node. The newer version is allowed to join the older cluster.

>     * Can we also validate that a cluster is under going repair/streaming - and not allow an upgrade ?

No such requirement either. But the upgrade implies a restart of scylla server. The streaming with this node get restarted will fail. 

> 
> 
> thoughts ?

**Comment by asias** [comment]
With scylla-debuginfo-2.2.0-0.20180705.240b9f122.el7.x86_64.rpm, I am able to decode the two backtraces. They are related to compaction. @raphaelsc Any ideas? Is this an issue fixed by later 2.2.X branch?


```
$ rpm2cpio scylla-debuginfo-2.2.0-0.20180705.240b9f122.el7.x86_64.rpm | cpio -idmv
$ addr2line -Cfpi -e  ./usr/lib/debug/usr/bin/scylla.debug
0x00000000005b010c
0x00000000005b03b5
0x00000000005b0403
/lib64/libpthread.so.0+0x000000000000f36f
0x000000000183b9ae
0x00000000018b9b32
0x0000000000e0f2bd
0x00000000010e6293
0x00000000010e8d1e
0x00000000005b3c6b
0x00000000005916cb
0x000000000058e984
0x0000000000658c6e
0x000000000065f593
0x000000000072eaed
/lib64/libpthread.so.0+0x0000000000007dc4
/lib64/libc.so.6+0x00000000000f776c
seastar::print_with_backtrace(seastar::backtrace_buffer&) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:303
 (inlined by) print_with_backtrace at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:324
seastar::print_with_backtrace(char const*) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:331
seastar::install_oneshot_signal_handler<11, &seastar::sigsegv_action>()::{lambda(int, siginfo_t*, void*)#1}::_FUN(int, siginfo_t*, void*) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3797
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3783
 (inlined by) _FUN at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3779
?? ??:0
sstables::size_tiered_backlog_tracker::backlog(std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_write_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_write_progress_manager*> > > const&, std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_read_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_read_progress_manager*> > > const&) const at /usr/src/debug/scylla-2.2.0/sstables/compaction_strategy.cc:428
 (inlined by) sstables::size_tiered_backlog_tracker::backlog(std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_write_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_write_progress_manager*> > > const&, std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_read_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_read_progress_manager*> > > const&) const at /usr/src/debug/scylla-2.2.0/sstables/compaction_strategy.cc:452
compaction_backlog_manager::backlog() const at /usr/src/debug/scylla-2.2.0/sstables/compaction_manager.cc:612
 (inlined by) compaction_backlog_manager::backlog() const at /usr/src/debug/scylla-2.2.0/sstables/compaction_manager.cc:666
std::_Function_handler<double (), database::database(db::config const&, database_config)::{lambda()#4}>::_M_invoke(std::_Any_data const&) at /usr/src/debug/scylla-2.2.0/sstables/compaction_manager.hh:208
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/database.cc:2145
 (inlined by) _M_invoke at /opt/scylladb/include/c++/7/bits/std_function.h:301
std::_Function_handler<float (), std::function<double ()> >::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) std::_Function_handler<float (), std::function<double ()> >::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:301
std::_Function_handler<void (), backlog_controller::backlog_controller(seastar::scheduling_group, seastar::io_priority_class const&, std::chrono::duration<long, std::ratio<1l, 1000l> >, std::vector<backlog_controller::control_point, std::allocator<backlog_controller::control_point> >, std::function<float ()>)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/database.cc:2164
 (inlined by) backlog_controller::backlog_controller(seastar::scheduling_group, seastar::io_priority_class const&, std::chrono::duration<long, std::ratio<1l, 1000l> >, std::vector<backlog_controller::control_point, std::allocator<backlog_controller::control_point> >, std::function<float ()>)::{lambda()#1}::operator()() const at /usr/src/debug/scylla-2.2.0/backlog_controller.hh:79
 (inlined by) std::_Function_handler<void (), backlog_controller::backlog_controller(seastar::scheduling_group, seastar::io_priority_class const&, std::chrono::duration<long, std::ratio<1l, 1000l> >, std::vector<backlog_controller::control_point, std::allocator<backlog_controller::control_point> >, std::function<float ()>)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:316
std::_Function_handler<void (), seastar::reactor::run()::{lambda()#5}>::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:634
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3006
 (inlined by) _M_invoke at /opt/scylladb/include/c++/7/bits/std_function.h:316
seastar::reactor::signals::poll_signal() at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::reactor::signals::poll_signal() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:250
std::_Function_handler<bool (), seastar::reactor::run()::{lambda()#7}>::_M_invoke(std::_Any_data const&) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3155
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3051
 (inlined by) _M_invoke at /opt/scylladb/include/c++/7/bits/std_function.h:302
seastar::reactor::run() at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3076
seastar::smp::configure(boost::program_options::variables_map)::{lambda()#3}::operator()() const at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:4002
seastar::posix_thread::start_routine(void*) at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::posix_thread::start_routine(void*) at /usr/src/debug/scylla-2.2.0/seastar/core/posix.cc:52
?? ??:0
?? ??:0


$ addr2line -Cfpi -e ./usr/lib/debug/usr/bin/scylla.debug
0x00000000005b010c
0x00000000005b03b5
0x00000000005b0463
/lib64/libpthread.so.0+0x000000000000f36f
/lib64/libc.so.6+0x00000000000351d6
/lib64/libc.so.6+0x00000000000368c7
/lib64/libc.so.6+0x000000000002e145
/lib64/libc.so.6+0x000000000002e1f1
0x0000000000eb50b8
0x000000000183b9b0
0x00000000018b9b32
0x0000000000e0f2bd
0x00000000010e6293
0x00000000010e8d1e
0x00000000005b3c6b
0x00000000005916cb
0x000000000058e984
0x0000000000658c6e
0x000000000065f593
0x000000000072eaed
/lib64/libpthread.so.0+0x0000000000007dc4
/lib64/libc.so.6+0x00000000000f776c
seastar::print_with_backtrace(seastar::backtrace_buffer&) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:303
 (inlined by) print_with_backtrace at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:324
seastar::print_with_backtrace(char const*) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:331
seastar::install_oneshot_signal_handler<6, &seastar::sigabrt_action>()::{lambda(int, siginfo_t*, void*)#1}::_FUN(int, siginfo_t*, void*) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3801
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3783
 (inlined by) _FUN at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3779
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0

seastar::continuation<seastar::future<>::then_wrapped<column_family::seal_active_streaming_memtable_immediate(flush_permit&&)::{lambda()#1}::operator()()::{lambda()#1}::operator()()::{lambda(seastar::future<>)#2}, seastar::future<> >(column_family::seal_active_streaming_memtable_immediate(flush_permit&&)::{lambda()#1}::operator()()::{lambda()#1}::operator()()::{lambda(seastar::future<>)#2}&&)::{lambda(column_family::seal_active_streaming_memtable_immediate(flush_permit&&)::{lambda()#1}::operator()()::{lambda()#1}::operator()()::{lambda(seastar::future<>)#2})#1}>::run_and_dispose() at /usr/src/debug/scylla-2.2.0/seastar/core/future.hh:345                                                                                
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/seastar/core/future.hh:539
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/seastar/core/future.hh:509
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/seastar/core/shared_future.hh:264
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/database.cc:932
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/seastar/core/future.hh:1307
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/seastar/core/future.hh:1355
 (inlined by) operator()<seastar::future_state<> > at /usr/src/debug/scylla-2.2.0/seastar/core/future.hh:984
 (inlined by) run_and_dispose at /usr/src/debug/scylla-2.2.0/seastar/core/future.hh:413
sstables::size_tiered_backlog_tracker::backlog(std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_write_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_write_progress_manager*>
> > const&, std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_read_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_read_progress_manager*> > > const&) const at /usr/src/debug/scylla-2.2.0/sstables/compaction_strategy.cc:428
 (inlined by) sstables::size_tiered_backlog_tracker::backlog(std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_write_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_write_progress_manager*> > > const&, std::unordered_map<seastar::lw_shared_ptr<sstables::sstable>, backlog_read_progress_manager*, std::hash<seastar::lw_shared_ptr<sstables::sstable> >, std::equal_to<seastar::lw_shared_ptr<sstables::sstable> >, std::allocator<std::pair<seastar::lw_shared_ptr<sstables::sstable> const, backlog_read_progress_manager*> > > const&) const at /usr/src/debug/scylla-2.2.0/sstables/compaction_strategy.cc:452
compaction_backlog_manager::backlog() const at /usr/src/debug/scylla-2.2.0/sstables/compaction_manager.cc:612
 (inlined by) compaction_backlog_manager::backlog() const at /usr/src/debug/scylla-2.2.0/sstables/compaction_manager.cc:666
std::_Function_handler<double (), database::database(db::config const&, database_config)::{lambda()#4}>::_M_invoke(std::_Any_data const&) at /usr/src/debug/scylla-2.2.0/sstables/compaction_manager.hh:208
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/database.cc:2145
 (inlined by) _M_invoke at /opt/scylladb/include/c++/7/bits/std_function.h:301
std::_Function_handler<float (), std::function<double ()> >::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) std::_Function_handler<float (), std::function<double ()> >::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:301
std::_Function_handler<void (), backlog_controller::backlog_controller(seastar::scheduling_group, seastar::io_priority_class const&, std::chrono::duration<long, std::ratio<1l, 1000l> >, std::vector<backlog_controller::control_point, std::allocator<backlog_controller::control_point> >, std::function<float ()>)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at
/opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/database.cc:2164
 (inlined by) backlog_controller::backlog_controller(seastar::scheduling_group, seastar::io_priority_class const&, std::chrono::duration<long, std::ratio<1l, 1000l> >, std::vector<backlog_controller::control_point, std::allocator<backlog_controller::control_point> >, std::function<float ()>)::{lambda()#1}::operator()() const at /usr/src/debug/scylla-2.2.0/backlog_controller.hh:79
 (inlined by) std::_Function_handler<void (), backlog_controller::backlog_controller(seastar::scheduling_group, seastar::io_priority_class const&, std::chrono::duration<long, std::ratio<1l, 1000l> >, std::vector<backlog_controller::control_point, std::allocator<backlog_controller::control_point> >, std::function<float ()>)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:316
std::_Function_handler<void (), seastar::reactor::run()::{lambda()#5}>::_M_invoke(std::_Any_data const&) at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) ?? at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:634
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3006
 (inlined by) _M_invoke at /opt/scylladb/include/c++/7/bits/std_function.h:316
seastar::reactor::signals::poll_signal() at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::reactor::signals::poll_signal() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:250
std::_Function_handler<bool (), seastar::reactor::run()::{lambda()#7}>::_M_invoke(std::_Any_data const&) at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3155
 (inlined by) operator() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3051
 (inlined by) _M_invoke at /opt/scylladb/include/c++/7/bits/std_function.h:302
seastar::reactor::run() at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:3076
seastar::smp::configure(boost::program_options::variables_map)::{lambda()#3}::operator()() const at /usr/src/debug/scylla-2.2.0/seastar/core/reactor.cc:4002
seastar::posix_thread::start_routine(void*) at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::posix_thread::start_routine(void*) at /usr/src/debug/scylla-2.2.0/seastar/core/posix.cc:52
?? ??:0
?? ??:0
?? ??:0
```

**Comment by glommer** [comment]
Backtrace looks like a dup of https://github.com/scylladb/scylla/issues/3732

**Comment by asias** [comment]
The patch for #3732: "remove monitor if sstable write failed" is not in latest 2.2 branch

**Comment by asias** [comment]
@slivne Can we close it as a duplication of #3723? The issue is fixed with commit bd85071743301eed39e660a667347afea00f66ef.
